cmd/chronograf-migrator/dashboard.go

```diff
@@ -0,0 +1,397 @@
+		m[k] = influxdb.Axis{
+			Bounds: v.Bounds,
+			Label:  v.Label,
+			Prefix: v.Prefix,
+			Suffix: v.Suffix,
+			Base:   v.Base,
+			Scale:  v.Scale,
+		}
+	}
+
+	if _, exists := m["x"]; !exists {
+		m["x"] = influxdb.Axis{}
+	}
+	if _, exists := m["y"]; !exists {
+		m["y"] = influxdb.Axis{}
+	}
+
+	return m
+}
+
+func convertLegend(l chronograf.Legend) influxdb.Legend {
+	return influxdb.Legend{
+		Type:        l.Type,
+		Orientation: l.Orientation,
+	}
+}
+
+func convertColors(cs []chronograf.CellColor) []influxdb.ViewColor {
+	vs := []influxdb.ViewColor{}
+
+	hasTextColor := false
+	hasThresholdColor := false
+	for _, c := range cs {
+		if c.Type == "text" {
+			hasTextColor = true
+		}
+		if c.Type == "threshold" {
+			hasThresholdColor = true
+		}
+
+		v := influxdb.ViewColor{
+			ID:   c.ID,
+			Type: c.Type,
+			Hex:  c.Hex,
+			Name: c.Name,
+		}
+		vs = append(vs, v)
+	}
+
+	if !hasTextColor {
+		vs = append(vs, influxdb.ViewColor{
+			ID:    "base",
+			Type:  "text",
+			Hex:   "#00C9FF",
+			Name:  "laser",
+			Value: 0,
+		})
+	}
+
+	if !hasThresholdColor {
+		vs = append(vs, influxdb.ViewColor{
+			ID:    "t",
+			Type:  "threshold",
+			Hex:   "#4591ED",
+			Name:  "ocean",
+			Value: 80,
+		})
+	}
+
+	return vs
+}
+
+var influxQLVarPattern = regexp.MustCompile(`'?:(\w+):'?`)
+
+func transpileQuery(q string) (string, error) {
+	now := time.Now()
+	t := influxql.NewTranspilerWithConfig(dbrpMapper{}, influxql.Config{
+		Now:            now,
+		FallbackToDBRP: true,
+	})
+
+	query := q
+	query = strings.Replace(query, ":interval:", "8675309ns", -1)
+	query = strings.Replace(query, ":dashboardTime:", "now() - 15m", 1)
+	query = strings.Replace(query, ":upperDashboardTime:", "now()", 1)
+
+	// TODO(desa): replace all variables not using this hack
+	query = influxQLVarPattern.ReplaceAllString(query, "'$1'")
+
+	pkg, err := t.Transpile(context.Background(), query)
+	if err != nil {
+		return "", err
+	}
+
+	return ast.Format(pkg), nil
+}
+
+func convertQueries(qs []chronograf.DashboardQuery) []influxdb.DashboardQuery {
+
+	ds := []influxdb.DashboardQuery{}
+	for _, q := range qs {
+		queryText := q.Command
+		if q.Type == "influxql" {
+			// if the query is influxql, add it as a comment and attempt to
+			// compile it to flux
+			queryText = fmt.Sprintf("// %s", queryText)
+
+			tq, err := transpileQuery(q.Command)
+			if err != nil {
+				queryText = fmt.Sprintf("// Failed to transpile query: %v\n%s", err, queryText)
+			} else {
+				queryText = fmt.Sprintf("// Original Query:\n%s\n\n%s", queryText, tq)
+			}
+		}
+
+		d := influxdb.DashboardQuery{
+			Text:     queryText,
+			EditMode: "advanced",
+		}
+
+		ds = append(ds, d)
+	}
+
+	if len(ds) == 0 {
+		d := influxdb.DashboardQuery{
+			Text:     "// cell had no queries",
+			EditMode: "advanced",
+			BuilderConfig: influxdb.BuilderConfig{
+				// TODO(desa): foo
+				Buckets: []string{"bucket"},
+			},
+		}
+		ds = append(ds, d)
+	}
+
+	return ds
+}
+
+type dbrpMapper struct {
```

Comment:
does the nil value not suffice for this type? An odd dep 🤔

Reply:
it's a placeholder for when we need to specify how to turn 1.x dbrps into buckets. `nil` would probably work right now.